client/login/wp-login/index.jsx

```diff
@@ -104,12 +118,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( this.state.loaded && window.history.length > 1 ) {
```

Comment:
I've tried this before, but React throws an error in the console:
`Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) " data-reactid="41"><svg class="gridicon
 (server) " data-reactid="41">Email me a login lin`

The current method is based on @yurynix's [suggestion](https://github.com/Automattic/wp-calypso/pull/13786#discussion_r115421298).